website/blog/2023-4-10-incrementals-pt2.md

```diff
@@ -0,0 +1,538 @@
+---
+slug: incremental-backups-pt2
+title: "Incremental Microsoft 365 backups in action"
+description: "In part 1 we discussed how there’s more than one way to run backups, and how full and incremental backups differ. With all the background information out of the way, it’s time to see how incremental backups actually come together in Corso. To do this, we’ll discuss things in the context of a running example."
+authors: amartinez
+tags: [corso, microsoft 365, backups]
+date: 2023-04-10
+image: ./images/incremental-scale.png
+---
+<!-- vale Vale.Spelling = NO -->
+
+![diagram of an incremental scale encoder By Lambtron - Own work, CC BY-SA 4.0, https://commons.wikimedia.org/w/index.php?curid=81494644](./images/incremental-scale.png)
+<!-- vale Vale.Spelling = YES -->
+
+In [part 1](https://corsobackup.io/blog/incremental-backups-pt1/) we discussed how there’s more than
+one way to run backups, and how full and incremental backups differ.
+
+With all the background information out of the way, it’s time to see how
+incremental backups actually come together in Corso. To do this, we’ll discuss
+things in the context of a running example.
+<!-- truncate -->
+## Part 1: Starting state
+
+Let’s say that Corso has previously run a backup for a user’s OneDrive. At the
+time of the backup, the drive had a layout like the below when listing all items
+in the root folder:
+
+```JSON
+- folder1 (directory, ID: 1)
+  - super secret file1.txt (file, ID: 2)
+  - temp.log (file, ID: 3)
+- folder2 (directory, ID: 4)
+  - secret file.docx (file, ID: 5)
+- folder3 (directory, ID: 6)
+  - static file.docx (file, ID: 7)
+- folder4 (directory, ID: 8)
+  - plain file.txt (file, ID: 9)
+  - folder5 (directory, ID: 10)
+    - update.log (file, ID: 11)
+```
+
+The corresponding Corso backup would have the following items in the kopia item
+data snapshot (some prefix folders elided for brevity and file/folder names use
+directly for clarity):
+
+```JSON
+- onedrive
+  - root
+    - folder1
+      - super secret file1.txt
+      - temp.log
+    - folder2
+      - secret file.docx
+    - folder3
+      - static file.docx
+    - folder4
+      - plain file.txt
+      - folder5
+        - update.log
+- onedriveMetadata
+  - folderMap.json (provides mapping of folder IDs to paths)
+  - delta.json (provides delta tokens for endpoints)
+```
+
+Since this post is all about fetching only the changes since the last backup,
+let’s also assume that between the time of the last backup and now the user has
+done the equivalent to the following commands in their OneDrive:
+
+```bash
+rm -rf root/folder2 (directory subtree deletion)
```

Comment:
I think this is OK but would have been nice to use Windows PS or cmd commands.

Reply:
I have no preference if something else would work better. I just happened to use linux commands cause that's what I default to 😅

Reply:
Will use PS commands in my next tutorial, I promise :)